rs/index/src/collection/collection.rs

```diff
@@ -737,14 +737,21 @@ impl<Q: Quantizer + Clone + Send + Sync + 'static> Collection<Q> {
             .clone();
         match segment {
             BoxedImmutableSegment::PendingSegment(pending_segment) => {
-                let mut pending_segment = pending_segment.upgradable_read();
-                optimizer.optimize(&pending_segment)?;
-                pending_segment.build_index()?;
-                pending_segment.with_upgraded(|pending_segment_write| {
-                    pending_segment_write.apply_pending_deletions()?;
-                    pending_segment_write.switch_to_internal_index();
-                    Ok(())
-                })?;
+                let temp_storage_dir;
+                {
+                    let mut pending_segment = pending_segment.upgradable_read();
+                    temp_storage_dir = pending_segment.temp_invalidated_ids_storage.read().base_directory.clone();
```

Comment:
don't expose the member variable like this. create a function `get_temp_invalidated_storage_dir()` or something